src/panels/lovelace/views/hui-view.ts

```diff
@@ -263,12 +263,38 @@ export class HUIView extends LitElement {
 
     const elements: HUIView["_badges"] = [];
     // It's possible that a null value was stored as a badge entry
-    const badges = processConfigEntities(config.badges.filter(Boolean));
+    const badges = processConfigEntities(
+      config.badges.filter((badgeConf) => {
+        if (typeof badgeConf === "object") {
+          const stateObj = this.hass!.states[badgeConf.entity];
+
+          if (!stateObj) {
+            return false;
+          }
+
+          if (badgeConf.state_filter) {
+            for (const filter of badgeConf.state_filter) {
+              if (evaluateFilter(stateObj, filter)) {
+                return true;
+              }
+            }
+          } else {
+            return true;
+          }
+
+          return false;
+        }
+
+        return true;
+      })
+    );
+
     for (const badge of badges) {
-      const element = document.createElement("ha-state-label-badge");
       const entityId = badge.entity;
+      const stateObj = this.hass!.states[entityId];
+      const element = document.createElement("ha-state-label-badge");
       element.hass = this.hass;
-      element.state = this.hass!.states[entityId];
```

Comment:
Why was this changed?

Reply:
Some things moved around while testing, shouldn't have checked this part in :+1: